### ISSUE TITLE
(feature) vendor prefixing

### ISSUE THREAD

**Issue by cristianbote** [cristianbote/goober#9]
Figure out of this feature is needed

**Comment by imlinus** [comment]
If it's needed, then please have it made as a separate plugin,
I do love your goal of keeping it light <3

**Comment by cristianbote** [comment]
Hey @imlinus, that's a great idea! 😃 Thanks for your feedback!

**Comment by cristianbote** [comment]
Alrighty! 😄 

So, it's been a couple of fun weeks. I've rewritten the css parser and all the _under the hood_ compilation and I've managed to get the proper css support and generation - right now there are some corner cases, I believe.

With that in mind, what other libraries are doing, are using either https://github.com/thysultan/stylis.js or they make a case for post processing the statically extracted css. The part with statically extracted css, is really something that sounds great, but in the end it's not really. Max, the co-author of styled-components, has recently published this article(https://mxstbr.com/thoughts/css-in-js/) which goes into the exact issue.

So, bottom line, for now I think it's the user choice to include the proper vendor prefixing. If I were to add this feature, as you said, the size would increase dramatically and it'll be really hard to pick what to prefix or not.

Let me know what you think and feel free to open the ticket, again.

**Comment by robertwbradford** [comment]
Great library! I like the idea of a plugin or additional module which includes the option. We use currently use styled-components, and it is nice not to have to worry about adding them all over. 

Perhaps it could work with https://github.com/browserslist/browserslist to determine which prefixes are needed.

**Comment by cristianbote** [comment]
I think a separate, pluggable, package for `goober` might be something worth looking into. We were discussing adding the `styled.<tagName>` support as well, as a separate package, so maybe this could be created as well as a package.

I'll reopen it, and see where it leads.

**Comment by robertwbradford** [comment]
Sounds good!

<my-two-cents>
  As for the `styled.<tagName>` syntax, while it is nice, it does start to add weight to the library. For me using `styled("<tagName>")` would be just fine and it provides a single interface that works for HTML element names, React components, and custom elements.
</my-two-cents>

**Comment by JoviDeCroock** [comment]
This should be implemented now for v2, I'll add some integration tests this weekend.
